appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/StatefulSessionContainer.java

```diff
@@ -3191,26 +2907,23 @@ public void trimEvent(Object primaryKey, Object context) {
 
     private class ASyncPassivator implements Runnable {
 
+        @Override
         public void run() {
             final Thread currentThread = Thread.currentThread();
-            final ClassLoader previousClassLoader =
-                    currentThread.getContextClassLoader();
+            final ClassLoader previousClassLoader = currentThread.getContextClassLoader();
             final ClassLoader myClassLoader = loader;
-
             boolean decrementedTaskCount = false;
             try {
                 // We need to set the context class loader for
                 // this (deamon) thread!!
                 if (System.getSecurityManager() == null) {
                     currentThread.setContextClassLoader(myClassLoader);
                 } else {
-                    java.security.AccessController.doPrivileged
-                            (new java.security.PrivilegedAction() {
-                                public java.lang.Object run() {
-                                    currentThread.setContextClassLoader(myClassLoader);
-                                    return null;
-                                }
-                            });
+                    PrivilegedAction<Void> action = () -> {
+                        currentThread.setContextClassLoader(myClassLoader);
+                        return null;
+                    };
```

Comment:
Shouldn't that also be possible as 

```java
PrivilegedAction<Void> action = () -> currentThread.setContextClassLoader(myClassLoader);
```
?

Reply:
Nope, because the action must have a return type. I tried that again now ...
But we could make a special method in some utility module.

Reply:
ah yes, now I remember. the void ugliness :D